jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/ProjectEnricherTest.java

```diff
@@ -0,0 +1,48 @@
+package org.eclipse.jkube.enricher.generic.openshift;
+
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.openshift.api.model.Project;
+import org.eclipse.jkube.kit.common.JavaProject;
+import org.eclipse.jkube.kit.common.KitLogger;
+import org.eclipse.jkube.kit.config.resource.PlatformMode;
+import org.eclipse.jkube.kit.config.resource.ResourceConfig;
+import org.eclipse.jkube.kit.enricher.api.EnricherContext;
+import org.eclipse.jkube.kit.enricher.api.JKubeEnricherContext;
+import org.junit.Test;
+
+import java.util.Properties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.*;
+
+public class ProjectEnricherTest {
+
+    private EnricherContext context;
+
+    public void setExpectations(Properties properties, ResourceConfig resourceConfig) {
+        context = JKubeEnricherContext.builder()
+                .log(new KitLogger.SilentLogger())
+                .resources(resourceConfig)
+                .project(JavaProject.builder()
+                        .properties(properties)
+                        .groupId("group")
+                        .artifactId("artifact-id")
+                        .build())
+                .build();
+    }
+
+    @Test
+    public void convertNamespaceResourceToProject() {
+        // Given
+        Properties properties = new Properties();
+        properties.put("jkube.enricher.jkube-namespace.namespace", "example");
+        setExpectations(properties, new ResourceConfig());
+        final KubernetesListBuilder klb = new KubernetesListBuilder();
+        // When
+        new ProjectEnricher((JKubeEnricherContext) context).create(PlatformMode.openshift, klb);
+        // Then
+        assertThat(klb.build().getItems()).isEmpty();
```

Comment:
This test doesn't seem to be verifying what it's named. You need to test this scenario:

1. KubernetesListBuilder contains a Namespace object
2. When ProjectEnricher.create() is invoked
3. KubernetesListBuilder should contain Project object

Translating this to code should look like this:
```java
        final KubernetesListBuilder klb = new KubernetesListBuilder();
        klb.addToItems(new NamespaceBuilder().withNewMetadata().withName("foo").endMetadata());
        // When
        new ProjectEnricher((JKubeEnricherContext) context).create(PlatformMode.openshift, klb);
        // Then
        assertThat(klb.build())
            .extracting(KubernetesList::getItems)
            .asList()
            .hasSize(1)
            .first()
            .isInstanceOf(Project.class)
            .hasFieldOrPropertyWithValue("metadata.name", "foo");
```

Reply:
I think we should also add given KubernetesListBuilder is empty then no project will be created. What do you think, will that help in increasing coverage for ProjectEnricher?

Reply:
Sounds good to me.